daemon/cmd/endpoint.go

```diff
@@ -306,24 +306,20 @@ func (m *endpointCreationManager) DebugStatus() (output string) {
 // createEndpoint attempts to create the endpoint corresponding to the change
 // request that was specified.
 func (d *Daemon) createEndpoint(ctx context.Context, owner regeneration.Owner, epTemplate *models.EndpointChangeRequest) (*endpoint.Endpoint, int, error) {
-	if option.Config.EnableEndpointRoutes {
-		if epTemplate.DatapathConfiguration == nil {
-			epTemplate.DatapathConfiguration = &models.EndpointDatapathConfiguration{}
+	if epTemplate.DatapathConfiguration == nil {
+		dpConfig := endpoint.NewDatapathConfiguration()
+		epTemplate.DatapathConfiguration = &dpConfig
+	} else {
```

Comment:
Doesn't this mean that endpoint routes mode is only ever applied if the caller specifies the `epTemplate.DatapathConfiguration`? Why isn't this broken?

Reply:
Yes, that's my understanding as well :thinking: 

Not sure why it doesn't break in tests. I've set a reminder to look tomorrow (or at least open an issue if I don't get time to debug).

Reply:
I'm still unsure why it's not breaking anything in CI. I suspected that maybe we never received nil `epTemplate.DatapathConfiguration`s but checked the client's code and we should receive it. I even went so far as to [reproduce the marshalling bits](https://play.golang.org/p/T-IGW5Nt7UH) to double check it doesn't turn nil into an empty structure. It doesn't.

If we were hitting this, the agent would consider endpoint routes to be disabled for all new endpoints added via the API AFAICS. Looking at a couple tests with endpoint routes enabled, that's clearly not the case (I see the expected routes and BPF programs).

It's still a bug so I'll send a PR to fix, but we're missing something here :thinking: